src/libpriv/rpmostree-core.cxx

```diff
@@ -4464,6 +4462,12 @@ rpmostree_context_assemble (RpmOstreeContext *self, GCancellable *cancellable, G
         return FALSE;
     }
 
+  /* Remove duplicated tmpfiles entries.
+   * See https://github.com/coreos/rpm-ostree/issues/26
+   */
+  if (overlays->len > 0 || overrides_remove->len > 0 || overrides_replace->len > 0)
```

Comment:
@jlebon make a little tweak about this as checking that the remove does not trigger this, so move this to the outer.

After remove `chrony`, `/usr/lib/rpm-ostree/tmpfiles.d/chrony.conf ` is removed, but `rpm-ostree-autovar.conf` does not update.
```
[root@cosa-devsh ~]# rpm-ostree override remove chrony
[root@cosa-devsh ~]# ostree ls ostree/1/1/0 /usr/lib/rpm-ostree/tmpfiles.d/chrony.conf
error: Inspecting path '/usr/lib/rpm-ostree/tmpfiles.d/chrony.conf': No such file or directory: /usr/lib/rpm-ostree/tmpfiles.d/chrony.conf

[root@cosa-devsh ~]# ostree cat ostree/1/1/0 /usr/lib/tmpfiles.d/rpm-ostree-autovar.conf > new
[root@cosa-devsh ~]# grep chrony  new
d /var/lib/chrony 0750 chrony chrony - -
d /var/log/chrony 0750 chrony chrony - -
```

Reply:
Yes sorry, this is exactly what I meant in my earlier comment! :)